File: api/schemas/carriers.py
"""Carrier request/response schemas."""

from typing import Optional

from pydantic import BaseModel, Field


class CarrierCreate(BaseModel):
    carrier_id: str = Field(..., min_length=1)
    name: str = Field(..., min_length=1)
    inner_length_mm: float = Field(..., gt=0)
    inner_width_mm: float = Field(..., gt=0)
    inner_height_mm: float = Field(..., gt=0)
    max_weight_kg: float = Field(..., gt=0)
    is_active: bool = True
    priority: Optional[int] = Field(default=None, ge=1)


class CarrierUpdate(BaseModel):
    name: Optional[str] = None
    inner_length_mm: Optional[float] = Field(default=None, gt=0)
    inner_width_mm: Optional[float] = Field(default=None, gt=0)
    inner_height_mm: Optional[float] = Field(default=None, gt=0)
    max_weight_kg: Optional[float] = Field(default=None, gt=0)
    is_active: Optional[bool] = None
    priority: Optional[int] = Field(default=None, ge=1)


class CarrierResponse(BaseModel):
    id: str
    carrier_id: str
    name: str
    inner_length_mm: float
    inner_width_mm: float
    inner_height_mm: float
    max_weight_kg: float
    is_predefined: bool
    is_active: bool
    priority: Optional[int]

File: frontend/src/api/carriers.ts
import client from './client'

export interface Carrier {
  id: string
  carrier_id: string
  name: string
  inner_length_mm: number
  inner_width_mm: number
  inner_height_mm: number
  max_weight_kg: number
  is_predefined: boolean
  is_active: boolean
  priority: number | null
}

export interface CarrierCreate {
  carrier_id: string
  name: string
  inner_length_mm: number
  inner_width_mm: number
  inner_height_mm: number
  max_weight_kg: number
  is_active?: boolean
  priority?: number | null
}

export const carriersApi = {
  list: () => client.get<Carrier[]>('/carriers'),
  create: (data: CarrierCreate) => client.post<Carrier>('/carriers', data),
  update: (carrierId: string, data: Partial<Carrier>) =>
    client.patch<Carrier>(`/carriers/${carrierId}`, data),
  delete: (carrierId: string) => client.delete(`/carriers/${carrierId}`),
}
